### ISSUE TITLE
"Parameter comment must end with a full stop"

### ISSUE THREAD

**Issue by ghost** [backdrop-ops/phpcs#3]
`phpcs` displayed the error "Parameter comment must end with a full stop" on line 717 of the following code:

![image](https://user-images.githubusercontent.com/2385329/221735101-5a574b98-a2b1-4fbb-82e9-3bf8bd974b47.png)

I don't think this is correct and it's not taking lists into account. Also, I don't see many references to full-stops/periods in the documentation standards, other than the first, summary line of a docblock. So is this actually a thing?

In any case, I don't think adding a full-stop at line 717 solves anything. So something else needs to change here...

**Comment by indigoxela** [comment]
Seems like you want to rework Backdrop\Sniffs\Commenting?

Sure, give it a stab. (FTR: the rule has been forked from https://github.com/pfrenssen/coder/blob/8.3.x/coder_sniffer/Drupal/Sniffs/Commenting/FunctionCommentSniff.php, but we can adapt it to our needs)

**Comment by yorkshire-pudding** [comment]
There is another comment scenario that it might be worth considering.
In settings.php there are several optional code lines that are commented out.  These generate phpcs errors as they don't end with a full stop. Perhaps it would be ok if they ended with `;`

**Comment by herbdool** [comment]
Yes, I agree with adding `;` but I think we should also add the backtick for quasi-code such as this example which is from an actual PR:

```
/**
 *   Examples:
 *    * db:/default/config_active
 *    * db:config_active
 *    * db://config_active
 *    * db:/config_store/config_stage
*/
```

**Comment by herbdool** [comment]
This might be what we should be using: https://github.com/backdrop-ops/phpcs/blob/a5aa03cdc79c6b7e6105ee8809f7689de43c68e1/Backdrop/Sniffs/Commenting/FunctionCommentSniff.php#L995

Wrap the code example in `@code/@endcode`.

**Comment by kiamlaluno** [comment]
I think the issue is that PHP_CodeSniffer expects a comment to always contains sentences, but a comment could also contain a list of items, code, or a phrase. In none of those cases, a period is necessary.
It could be fine to add a period at the end of a phrase, but not for the other cases.

**Comment by kiamlaluno** [comment]
This is reported also from the rulesets used for Drupal; I saw many PHP_CodeSniffer reports for contributed modules that included errors about a missing period at the end of a comment.  
I guess that either we have a rule that gives false positives, or we disable it.

Can at least the error be changed to a warning?

**Comment by indigoxela** [comment]
Closing because of lack of activity and no PR.
